pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -475,16 +475,24 @@ def _map_devices_to_accelerator(self, accelerator: str) -> bool:
         if self.devices is None:
             return False
         if accelerator == DeviceType.TPU and _TPU_AVAILABLE:
+            if self.devices == "auto":
+                self.devices = self._get_auto_device_count(DeviceType.TPU)
```

Comment:
What's the point of having this function? This could just be

```suggestion
                self.devices = TPUAccelerator.auto_device_count()
```

Or if the accelerator has been already set at this point.


```suggestion
                self.devices = self.accelerator.auto_device_count()
```

Reply:
`accelerator` won't be set by then. Updated!